src/wallet/wallet.cpp

```diff
@@ -2407,10 +2421,11 @@ CWallet::Balance CWallet::GetBalance(const int min_depth, bool avoid_reuse) cons
     {
         auto locked_chain = chain().lock();
         LOCK(cs_wallet);
+        std::set<uint256> trustedParents;
         for (const auto& entry : mapWallet)
         {
             const CWalletTx& wtx = entry.second;
-            const bool is_trusted{wtx.IsTrusted(*locked_chain)};
+            const bool is_trusted{wtx.IsTrusted(*locked_chain, trustedParents)};
```

Comment:
With the overloaded `IsTrusted` this could be reverted? Same below.

Reply:
Nope -- because I'm caching the trust across the entire wallet (outside the loop) rather than just a single call.

Reply:
Right, missed this is in a loop 🤦‍♂